pkg/controllers/streaming/processor_controller_test.go

```diff
@@ -103,70 +184,106 @@ func TestReconcile(t *testing.T) {
 		},
 		ShouldErr: true,
 		ExpectStatusUpdates: []runtime.Object{
-			processorStatusConditionsUnknown(testNamespace, testName),
+			processorGiven.
+				StatusConditions(
+					factories.Condition().Type("DeploymentReady").Unknown(),
+					factories.Condition().Type("Ready").Unknown(),
+					factories.Condition().Type("ScaledObjectReady").Unknown(),
+					factories.Condition().Type("StreamsReady").Unknown(),
+				).
+				Get(),
 		},
 	}, {
 		Name: "processor sidecar image not present in configMap",
 		Key:  types.NamespacedName{Namespace: testNamespace, Name: testName},
 		GivenObjects: []runtime.Object{
-			processor(testNamespace, testName),
-			configMap(testNamespace, processorImages),
+			processorGiven.Get(),
+			configMapGiven.
+				Mutate(func(configMap *corev1.ConfigMap) {
```

Comment:
the mutate function makes me a bit nervous: is it guaranteed that the mutation effects are isolated per test case?
Did you consider introducing something like `configMapWithoutProcSidecarImageGiven` (or a better name hopefully)? It could even be inlined: `factories.ConfigMap().Xxx`.

Reply:
Agreed.  The inlining works better.

Reply:
every operation on a factory creates and returns a deep copy of the state, they are intended to branch.

I generally avoided using `.Mutate()` directly in a test, favoring keeping objects less defined until the richer definition is needed.

Reply:
Replaced direct usages of `.Mutate()`.

Reply:
For the record, for some factories (e.g. `podTemplateSpecImpl`), `Mutate()` actually mutates the factory and for others (the majority), it returns a modified copy but leave the receiver unchanged. We probably need to look at this in another PR...